plugins/main/public/components/overview/malware-detection/dashboard/dashboard.tsx

```diff
@@ -0,0 +1,141 @@
+import { getDashboardPanels } from './dashboard-panels';
+import { I18nProvider } from '@osd/i18n/react';
+import useSearchBar from '../../../common/search-bar/use-search-bar';
+import './styles.scss';
+import { withErrorBoundary } from '../../../common/hocs';
+import { IndexPattern } from '../../../../../../../src/plugins/data/common';
+import {
+  ErrorFactory,
+  ErrorHandler,
+  HttpError,
+} from '../../../../react-services/error-management';
+import { compose } from 'redux';
+import { SampleDataWarning } from '../../../visualize/components';
+import {
+  AlertsDataSourceRepository,
+  PatternDataSource,
+  tParsedIndexPattern,
+  useDataSource,
+  AlertsMalwareDetectionDataSource,
+} from '../../../common/data-source';
+import { DiscoverNoResults } from '../../../common/no-results/no-results';
+import { LoadingSpinner } from '../../../common/loading-spinner/loading-spinner';
+
+const plugins = getPlugins();
+
+const SearchBar = getPlugins().data.ui.SearchBar;
+
+const DashboardByRenderer = plugins.dashboard.DashboardContainerByValueRenderer;
+
+const DashboardMalwareDetectionComponent: React.FC = ({}) => {
+  const {
+    filters,
+    dataSource,
+    fetchFilters,
+    isLoading: isDataSourceLoading,
+    fetchData,
+    setFilters,
+  } = useDataSource<tParsedIndexPattern, PatternDataSource>({
+    DataSource: AlertsMalwareDetectionDataSource,
+    repository: new AlertsDataSourceRepository(),
+  });
+
+  const [results, setResults] = useState<SearchResponse>({} as SearchResponse);
+
+  const { searchBarProps } = useSearchBar({
+    indexPattern: dataSource?.indexPattern as IndexPattern,
+    filters,
+    setFilters,
+  });
+
+  const { query, dateRangeFrom, dateRangeTo } = searchBarProps;
+
+  useEffect(() => {
+    if (isDataSourceLoading) {
+      return;
+    }
+    fetchData({
+      query,
+      dateRange: {
+        to: dateRangeTo,
+        from: dateRangeFrom,
+      },
+    })
+      .then(results => setResults(results))
+      .catch(error => {
+        const searchError = ErrorFactory.create(HttpError, {
+          error,
+          message: 'Error fetching data',
+        });
+        ErrorHandler.handleError(searchError);
+      });
+  }, [
+    JSON.stringify(fetchFilters),
+    JSON.stringify(query),
+    JSON.stringify(dateRangeFrom),
+    JSON.stringify(dateRangeTo),
+  ]);
+
+  return (
+    <>
+      <I18nProvider>
+        <>
+          {isDataSourceLoading && !dataSource ? (
+            <LoadingSpinner />
+          ) : (
+            <div className='wz-search-bar hide-filter-control'>
+              <SearchBar
+                appName='malware-detection-searchbar'
+                {...searchBarProps}
+                showDatePicker={true}
+                showQueryInput={true}
+                showQueryBar={true}
+              />
+            </div>
+          )}
+          <SampleDataWarning />
+          {dataSource && results?.hits?.total === 0 ? (
+            <DiscoverNoResults />
+          ) : null}
```

Comment:
If the sample data warning were added in the rendering of the condition to display the no results, the warning would always be shown as long as there are results, thus avoiding showing the sample data while no results are visible.

4.9.0 Dashboard

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/124377319/60773dcf-9d78-4d03-9e88-905a3e83de18)

Embeddable

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/124377319/f816989c-2625-4cd6-9510-1e3cb705b869)

Reply:
Resolved in the commit [94cd1a2](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6521/commits/94cd1a26cf6c031e19d43b9a8839cf5fd471cbc7)